src/redactableMarkdownParser.js

```diff
@@ -18,78 +19,89 @@ const tiplink = require('./plugins/parser/tiplink');
 const vocablink = require('./plugins/parser/vocablink');
 
 module.exports = class RedactableMarkdownParser {
-  static getPlugins = function() {
-    return this.getParserPlugins().concat(this.getCompilerPlugins());
-  }
-
-  static getParserPlugins = function() {
-    return [
-      restorationRegistration,
-      divclass,
-      redactedLink,
-      resourcelink,
-      tip,
-      tiplink,
-      vocablink,
-    ];
-  };
-
-  static getCompilerPlugins = function() {
-    return [
-      indent,
-      rawtext,
-    ]
-  }
 
-  static getParser = function() {
-    return unified()
+  constructor() {
+    this.parser = unified()
       .use(parse, {
         commonmark: true,
         pedantic: true
-      })
-      .use(this.getParserPlugins());
-  };
+      }).use(this.constructor.getParserPlugins());
+  }
 
-  static sourceToHtml = function(source) {
+  loadPlugins(pluginPaths) {
+    pluginPaths.split(/,/).forEach((pluginPath) => {
+      const plugin = require(path.resolve(process.cwd(), pluginPath));
+      this.parser.use(plugin);
+    });
+  }
+
+  getParser() {
+    return this.parser();
+  }
+
+  sourceToHtml(source) {
     return this.getParser()
       .use(html)
       .processSync(source)
       .contents;
-  };
+  }
 
-  static sourceToRedactedMdast = function(source) {
+  sourceToRedactedMdast(source) {
     return this.getParser()
       .use({ settings: { redact: true } })
       .parse(source);
-  };
+  }
 
-  static sourceToRedacted = function(source) {
+  sourceToRedacted(source) {
     const sourceTree = this.sourceToRedactedMdast(source);
     return this.getParser()
       .use(stringify)
       .use(renderRedactions)
       .stringify(sourceTree);
-  };
+  }
 
-  static sourceAndRedactedToMergedMdast = function(source, redacted) {
+  sourceAndRedactedToMergedMdast(source, redacted) {
     const sourceTree = this.sourceToRedactedMdast(source);
     const redactedTree = this.getParser()
       .use(restoreRedactions(sourceTree))
       .parse(redacted);
 
     return redactedTree;
-  };
+  }
 
-  static sourceAndRedactedToMarkdown = function(source, redacted) {
+  sourceAndRedactedToMarkdown(source, redacted) {
     const mergedMdast = this.sourceAndRedactedToMergedMdast(source, redacted);
     return this.getParser()
       .use(stringify)
-      .use(this.getCompilerPlugins())
+      .use(this.constructor.getCompilerPlugins())
       .stringify(mergedMdast);
-  };
+  }
 
-  static sourceAndRedactedToHtml = function(source, redacted) {
+  sourceAndRedactedToHtml(source, redacted) {
     const restoredMarkdown = this.sourceAndRedactedToMarkdown(source, redacted);
     return this.sourceToHtml(restoredMarkdown);
-  };
+  }
+
+  static getParserPlugins() {
+    return [
+      restorationRegistration,
+      divclass,
+      redactedLink,
+      resourcelink,
+      tip,
+      tiplink,
```

Comment:
I thought the whole idea of this change was to move things like "tiplink" out of this repo.  Is that going to be a follow-up change?

Reply:
Yep! That's in progress here https://github.com/code-dot-org/redactable-markdown/pull/40